pennylane/devices/qubit/simulate.py

```diff
@@ -124,24 +134,31 @@ def get_final_state(circuit, debugger=None, interface=None):
 
     # initial state is batched only if the state preparation (if it exists) is batched
     is_state_batched = bool(prep and prep.batch_size is not None)
+    measurement_values = {}
     for op in circuit.operations[bool(prep) :]:
+        if isinstance(op, Conditional):
+            meas_id = op.meas_val.measurements[0].hash
+            if not measurement_values[meas_id]:
```

Comment:
I thought it would make sense to error if `Conditional` depends on something that is undefined. I had manually introduced an error there, but I was not able to test for it, at least not in a meaningful way. I thus thought I would just rely on the standard `dict` errors. If `Conditional` should branch to False for undefined values however, let me know.